### PR TITLE
Introduce benchmark utilities

### PR DESCRIPTION
**Pull request by vwxyzjn**
## Description
This PR introduces utilities that help us to run the benchmark experiments more smoothly.

Previously, we relied on a very simple mechanism for conducting benchmark experiments such as 

https://github.com/vwxyzjn/cleanrl/blob/443bb141afd26db41f14ede4a5f2026a62a16756/benchmark/sac/mujoco.sh#L3-L7

Such bash script usage is pretty straightforward but lacks flexibility and configurability. This PR introduce a command such as 

```bash
OMP_NUM_THREADS=1 python -m cleanrl_utils.benchmark \
    --env-ids CartPole-v1 Acrobot-v1 MountainCar-v0 \
    --command "poetry run python cleanrl/ppo.py --track --cuda False" \
    --num-seeds 3 \
    --workers 3
```

which will automatically run experiments with`workers=3` subprocesses. A full example can be seen here:

```
# export WANDB_ENTITY=openrlbenchmark
# export WANDB_PROJECT=cleanrl

OMP_NUM_THREADS=1 python -m cleanrl_utils.benchmark \
    --env-ids CartPole-v1 Acrobot-v1 MountainCar-v0 \
    --command "poetry run python cleanrl/ppo.py --track --cuda False" \
    --num-seeds 3 \
    --workers 3

poetry install -E atari
python -m cleanrl_utils.benchmark \
    --env-ids PongNoFrameskip-v4 BeamRiderNoFrameskip-v4 BreakoutNoFrameskip-v4 \
    --command "poetry run python cleanrl/ppo_atari.py --track" \
    --num-seeds 3 \
    --workers 1

python -m cleanrl_utils.benchmark \
    --env-ids PongNoFrameskip-v4 BeamRiderNoFrameskip-v4 BreakoutNoFrameskip-v4 \
    --command "poetry run python cleanrl/ppo_atari_lstm.py --track" \
    --num-seeds 3 \
    --workers 1

python -m cleanrl_utils.benchmark \
    --env-ids PongNoFrameskip-v4 BeamRiderNoFrameskip-v4 BreakoutNoFrameskip-v4 \
    --command "poetry run python cleanrl/ppo_atari_envpool.py --track" \
    --num-seeds 3 \
    --workers 1

poetry install -E "mujoco pybullet"
python -c "import mujoco_py"
OMP_NUM_THREADS=1 python -m cleanrl_utils.benchmark \
    --env-ids HalfCheetah-v2 Walker2d-v2 Hopper-v2 \
    --command "poetry run python cleanrl/ppo_continuous_action.py --track --cuda False" \
    --num-seeds 3 \
    --workers 3

poetry install -E procgen
python -m cleanrl_utils.benchmark \
    --env-ids starpilot bossfight bigfish \
    --command "poetry run python cleanrl/ppo_procgen.py --track --track" \
    --num-seeds 3 \
    --workers 1
```

## Types of changes

- [ ] Bug fix
- [x] New feature
- [ ] New algorithm
- [ ] Documentation

## Checklist:

- [x] I've read the [CONTRIBUTION](https://github.com/vwxyzjn/cleanrl/blob/master/CONTRIBUTING.md) guide (**required**).
- [x] I have ensured `pre-commit run --all-files` passes (**required**).
- [ ] I have updated the documentation and previewed the changes via `mkdocs serve`.

### CODE REVIEW

**Review by dosssman — APPROVED** [review]
Quite useful addition. Will try to properly use it for the next benchmarks.

**Comment by vwxyzjn** [comment]
Added more documentation. Going to merge this soon.

File: cleanrl_utils/benchmark.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    print(commands)
+
+    if args.workers > 0:
+        from concurrent.futures import ThreadPoolExecutor
+
+        executor = ThreadPoolExecutor(max_workers=args.workers, thread_name_prefix="cleanrl-benchmark-worker-")
+        for command in commands:
+            executor.submit(run_experiment, command)
+        executor.shutdown(wait=True, cancel_futures=False)
```
Comment (yooceii) [inline cleanrl_utils/benchmark.py:45]: If it simply spawns multiple processes, why don't we instead write a bash script?

File: docs/get-started/benchmark-utility.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+```bash
+OMP_NUM_THREADS=1 xvfb-run -a python -m cleanrl_utils.benchmark \
+    --env-ids CartPole-v1 Acrobot-v1 MountainCar-v0 \
+    --command "poetry run python cleanrl/ppo.py --cuda False --track --capture-video" \
+    --num-seeds 3 \
+    --workers 5
+```
+
+What just happened here? In principle the helps run the following commands in 5 subprocesses:
```
Comment (yooceii) [inline docs/get-started/benchmark-utility.md:38]: Kinda confusing. what is "helps" mean?

File: cleanrl_utils/benchmark.py (hunk 1 above)
Comment (vwxyzjn) [reply cleanrl_utils/benchmark.py:45]: I’m open to using a bash script. But the main benefit here is pythons code is a bit easier to read and also allows me to set a maximum number of workers. I don’t know if I can Set a maximum number of workers with bash.

File: docs/get-started/benchmark-utility.md (hunk 2 above)
Comment (vwxyzjn) [reply docs/get-started/benchmark-utility.md:38]: Typo. In principle *the* script helps…

File: cleanrl_utils/benchmark.py (hunk 1 above)
Comment (yooceii) [reply cleanrl_utils/benchmark.py:45]: I think u can use `poetry run python cleanrl/ppo.py &` to spawn a new process and run it in the background.

Comment (vwxyzjn) [reply cleanrl_utils/benchmark.py:45]: True. The issue is setting a maximum amount of workers. Imagine I have 60 commands to run, `poetry run python cleanrl/ppo.py &` is just gonna overflow my CPU.

**Comment by vwxyzjn** [comment]
Merging now.

**Pull request merged**